src/Components/LifecycleTable/LifecycleTable.test.tsx

```diff
@@ -0,0 +1,654 @@
+      const dateA = new Date(a.end_date || '').getTime();
+      const dateB = new Date(b.end_date || '').getTime();
+      return direction === 'asc' ? dateA - dateB : dateB - dateA;
+    });
+  }),
+  filterChartDataBySystems: jest.fn((data, _lifecycleValue, direction) => {
+    return [...data].sort((a, b) => {
+      return direction === 'asc' ? a.count - b.count : b.count - a.count;
+    });
+  }),
+}));
+
+describe('LifecycleTable', () => {
+  const mockStreamData: Stream[] = [
+    {
+      name: 'nodejs',
+      application_stream_name: 'Node.js 18',
+      display_name: 'Node.js 18',
+      os_major: 9,
+      os_minor: 0,
+      start_date: '2023-01-01',
+      end_date: '2025-01-01',
+      count: 25,
+      rolling: false,
+      related: false,
+      support_status: 'Supported',
+      os_lifecycle: 'Supported',
+      systems_detail: [{ id: '343252f2f23f44g', display_name: 'Test App System 1' }],
+    },
+    {
+      name: 'python',
+      application_stream_name: 'Python 3.11',
+      display_name: 'Python 3.11',
+      os_major: 9,
+      os_minor: 0,
+      start_date: '2022-01-01',
+      end_date: '2024-01-01',
+      count: 30,
+      rolling: false,
+      related: false,
+      support_status: 'Supported',
+      os_lifecycle: 'Supported',
+      systems_detail: [{ id: '343252f2f23f44g', display_name: 'Test App System 1' }],
+    },
+    {
+      name: 'httpd',
+      application_stream_name: 'Apache HTTP Server',
+      display_name: 'Apache HTTP Server',
+      os_major: 8,
+      os_minor: 0,
+      start_date: '2022-01-01',
+      end_date: '2024-01-01',
+      count: 15,
+      rolling: false,
+      related: true,
+      support_status: 'Supported',
+      os_lifecycle: 'Supported',
+      systems_detail: [{ id: '10', display_name: 'Test App System 2' }],
+    },
+  ];
+
+  const mockSystemData: SystemLifecycleChanges[] = [
+    {
+      name: 'RHEL 9.3',
+      major: 9,
+      minor: 3,
+      start_date: '2023-01-01',
+      end_date: '2025-01-01',
+      count: 100,
+      lifecycle_type: 'standard',
+      related: false,
+      display_name: 'RHEL 9.3',
+      support_status: 'Supported',
+      systems_detail: [{ id: '23fr2f23rf42f0', display_name: 'Test System 1' }],
+    },
+    {
+      name: 'RHEL 8.9',
+      major: 8,
+      minor: 9,
+      start_date: '2022-01-01',
+      end_date: '2024-01-01',
+      count: 50,
+      lifecycle_type: 'standard',
+      related: true,
+      display_name: 'RHEL 8.9',
+      support_status: 'Supported',
+      systems_detail: [{ id: '23fr2f23rf42f0', display_name: 'Test System 1' }],
+    },
+  ];
+
+  const mockUpdateChartSortValue = jest.fn();
+  const defaultProps = {
+    updateChartSortValue: mockUpdateChartSortValue,
+    lifecycleDropdownValue: 'all',
+  };
+
+  // Helper function to render component with act wrapping
+  const renderWithAct = async (component: React.ReactElement) => {
+    let result: any;
+    await act(async () => {
```

Comment:
Just a note about `act`. In the IDE I saw a message about [deprecation](https://react.dev/warnings/react-dom-test-utils) of the `act`. I searched a bit more and I found it's probably a [bug](https://stackoverflow.com/questions/78438525/jest-tells-me-to-use-act-but-then-ide-indicates-it-is-deprecated-what-is-best) and using `act` should be fine. I tried upgrading to `@testing-library/react@15.0.6` (which is a recommended solution). The update introduces warnings in the output, which seems to be another [bug](https://github.com/testing-library/react-testing-library/issues/1413) across various versions.

Conclusion: I would keep the unit test as it is since it works :) (just keeping the note here)